core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -368,21 +378,21 @@ trait Endpoint[A] { self =>
  */
 object Endpoint {
 
-  type Result[A] = EndpointResult[A]
+  type Result[F[_], A] = EndpointResult[F, A]
 
   /**
    * Creates an empty [[Endpoint]] (an endpoint that never matches) for a given type.
    */
-  def empty[A]: Endpoint[A] = new Endpoint[A] {
-    final def apply(input: Input): Result[A] = EndpointResult.NotMatched
+  def empty[F[_] : Effect, A]: Endpoint[F, A] = new Endpoint[F, A] {
+    final def apply(input: Input): Result[F, A] = EndpointResult.NotMatched
   }
 
   /**
    * Creates an [[Endpoint]] that always matches and returns a given value (evaluated eagerly).
    */
-  def const[A](a: A): Endpoint[A] = new Endpoint[A] {
-    final def apply(input: Input): Result[A] =
-      EndpointResult.Matched(input, Trace.empty, Rerunnable.const(Output.payload(a)))
+  def const[F[_] : Effect, A](a: A): Endpoint[F, A] = new Endpoint[F, A] {
```

Comment:
That necessity of putting effect type together with value type is striking me a bit, so I was thinking if it's possible to provide syntax like `Endpoint[F].const(a)`, so `A` will be derived automatically.

Ideally, both types here are inferred automatically if there is only one implicit `Effect` in scope, but it's a bit unclear behavior.

Reply:
Oh yeah, we should do that. See my comment above about Iterant's builders.

Reply:
Yes, I like the idea